### PR TITLE
Add docs re: component translation

### LINKED ISSUE #514: Document how to add translations to a Component
Explain how the i18n Provider and translation wrapper work.

### PR DESCRIPTION
**Pull request by muffinresearch**
Fixes #514

### CODE REVIEW

File: docs/i18n.md [hunk 1]
```diff
@@ -84,3 +84,113 @@ the JSON to the repo.
...
+fancy chained API but we've stuck to a more traditional approach.
+
+Before we get into how to make use of these functions let's take a look at
+how the Jed instance is exposed to our components.
+
+### The Translation Provider
+
+The translation provider is used to pass down a Jed instance via context to
+components lower down in the component hierarchy. This part is already done
+for you in the addons-frontend code-base. So you should only need to worry
```
Comment (tofumatt) [inline docs/i18n.md]: "code-base" seems odd to me.

"This part is already done for you in addons-frontend" would be fine.

File: docs/i18n.md [hunk 2]
```diff
@@ -84,3 +84,113 @@ the JSON to the repo.
...
+about wrapping your components as detailed in the next section.
+
+## The translate component wrapper.
+
+The translate Higher Order Component is a helper that wraps any component
+and takes the `Jed` `i18n` instance from context and makes it  available in
```
Comment (tofumatt) [inline docs/i18n.md]: `Jed` isn't wrapped in <code>\`</code> elsewhere, probably doesn't need to be here (plus it makes the important `i18n` stand out more).

File: docs/i18n.md [hunk 3]
```diff
@@ -84,3 +84,113 @@ the JSON to the repo.
...
+the wrapped component's props.
+
+Here's an example of a very basic component setup for translation:
```
Comment (tofumatt) [inline docs/i18n.md]: Apparently I'm in editor mode this morning, but the "very" here seems superfluous. Maybe it's the table of non-English speakers next to me at the cafe, but I feel it just complicates the text 😄

File: docs/i18n.md [hunk 4]
```diff
@@ -84,3 +84,113 @@ the JSON to the repo.
...
+
+That's pretty much all there is to it.
+
+## Using the Jed API
+
+Once you have `i18n` available to your component you can then use
+any of the Jed methods exposed on the `i18n` object.
+
+```
+gettext = function ( key )
```
Comment (tofumatt) [inline docs/i18n.md]: This is so handy to have right in the translation docs, thanks! 👍

File: docs/i18n.md [hunk 5]
```diff
@@ -84,3 +84,113 @@ the JSON to the repo.
...
+```
+
+That's pretty much all there is to it.
+
+## Using the Jed API
+
+Once you have `i18n` available to your component you can then use
+any of the Jed methods exposed on the `i18n` object.
+
+```
```
Comment (tofumatt) [inline docs/i18n.md]: Could add the "javascript" text here to syntax highlight the functions below.

File: docs/i18n.md [hunk 6]
```diff
@@ -84,3 +84,113 @@ the JSON to the repo.
...
+dcngettext = function ( domain, singular_key, plural_key, value, category )
+pgettext = function ( context, key )
+dpgettext = function ( domain, context, key )
+npgettext = function ( context, singular_key, plural_key, value )
+dnpgettext = function ( domain, context, singular_key, plural_key, value )
+dcnpgettext = function ( domain, context, singular_key, plural_key, value, category )
+sprintf = function ( string, substitutions)
+```
+
+### Using sprintf
```
Comment (tofumatt) [inline docs/i18n.md]: Would be nice to wrap `sprintf` in <code>`</code>.

File: docs/i18n.md [hunk 7]
```diff
@@ -84,3 +84,113 @@ the JSON to the repo.
...
+
+Both of these approaches allow for translators to re-order the substitution
+vars.
+
+### Guidance on HTML in translations
+
+Generally we're looking to avoid having HTML in the middle of translation
+strings as much as possible.
+
+If you need HTML it's better to use substitutions to add the HTML than
```
Comment (tofumatt) [inline docs/i18n.md]: Could you include an example of this? It's not how we did it in Django if I recall correctly, and while it sounds safer it also sounds a lot more complicated in my head.

File: docs/i18n.md [hunk 8]
```diff
@@ -84,3 +84,113 @@ the JSON to the repo.
 # build the JSON.
 NODE_APP_INSTANCE=[MY_APP] bin/build-locales
 ```
+
+## Setting up translations
+
+To set-up a component to be translated there are two pieces of code to know
```
Comment (tofumatt) [inline docs/i18n.md]: I think technically it's just "set up".

File: docs/i18n.md (hunk 5 above)
Comment (muffinresearch) [inline docs/i18n.md]: It's more pseudo code lifted from the Jed docs, but I'll try it out...

Comment (tofumatt) [inline docs/i18n.md]: Oh, never mind then. It's no biggie.

**Comment by tofumatt** [comment]
r+wc

**Pull request merged**